### PR TITLE
resource_control: watch delete with prev and refine test

### LINKED ISSUE #7095: resource_control: can not delete resource groups by watch etcd
## Bug Report
`groupsController` is updated by watch channel, but we need to enable `WithPrevKV` to acquire delete value.
That will result In when occurring delete action `groupsController` will not updated by watch channel

### What did you do?
delete resource group

### What did you expect to see?
delete in `groupsController` as well

### What did you see instead?
can not delete in `groupsController`

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #7095

### What is changed and how does it work?

```commit-message
```
- `groupsController` is updated by watch channel, but we need to enable WithPrevKV to acquire delete value.
That will result In when occurring delete action groupsController will not updated by watch channel
- remove watch resource groups interface which is just used in test
- refine watch related test

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/resource_group/controller/controller.go [hunk 1]
```diff
@@ -277,19 +278,29 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
...
+							if err := proto.Unmarshal(item.PrevKv.Value, group); err != nil {
+								continue
+							}
+							if _, ok := c.groupsController.LoadAndDelete(group.Name); ok {
+								resourceGroupStatusGauge.DeleteLabelValues(group.Name)
+							}
+						} else {
+							// Prev-kv is compacted means there must have been a delete event before this event,
+							// which means that this is just a duplicated event, so we can just ignore it.
+							log.Info("previous key-value pair has been compacted", zap.String("previous key", string(item.Kv.Key)))
```
Comment (CabinfeverB) [inline client/resource_group/controller/controller.go:303]: Why is `previous key`? and `Kv.Key` will be nil?

**Review by CabinfeverB — COMMENTED** [review]
Could u add a test?
And in `client.go` there is also a implment of `Watch Resource Group`.

File: client/resource_group/controller/controller.go (hunk 1 above)
Comment (HuSharp) [reply client/resource_group/controller/controller.go]: fixed global config as well.

**Comment by HuSharp** [comment]
> Could u add a test? And in `client.go` there is also a implment of `Watch Resource Group`.

- remove `watchResourceGroups` interface which is just used in test.
- refine related watch test.
PTAL, thx!

**Review by CabinfeverB — APPROVED** [review]

**Comment by HuSharp** [comment]
@nolouch friendly ping :)

**Review by nolouch — APPROVED** [review]

**Pull request merged**